engines/sci/parser/vocabulary.cpp

```diff
@@ -482,6 +511,42 @@ void Vocabulary::lookupWord(ResultWordList& retval, const char *word, int word_l
 	}
 }
 
+void Vocabulary::lookupWordPrefix(ResultWordListList &parent_retval, ResultWordList &retval, const char *word, int word_len) {
+	// currently, this is needed only for Hebrew translation
+	if (g_sci->getLanguage() != Common::HE_ISR)
+		return;
+
+	if (--word_len <= 0)
+		return;
+
+	if (lookupSpecificPrefix(parent_retval, retval, word, word_len, 0xe1, "1hebrew1prefix1bet"))			// "Bet"
+		return;
+	if (lookupSpecificPrefix(parent_retval, retval, word, word_len, 0xe4, "the"))							// "He Hayedia"
+		return;
+	if (lookupSpecificPrefix(parent_retval, retval, word, word_len, 0xec, "1hebrew1prefix1lamed"))			// "Lamed"
+		return;
+	if (lookupSpecificPrefix(parent_retval, retval, word, word_len, 0xee, "1hebrew1prefix1mem"))			// "Mem"
+		return;
```

Comment:
You can put all of these prefixes into an array, and iterate through it, instead of repeating the same code